packages/headless/src/modules/user-auth/decorator/user.decorator.ts

```diff
@@ -0,0 +1,28 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { IJwtPayload } from '../interface/user.interface';
+
+/**
+ * @return only the userId from jwt payload
+ */
+export const UserId = createParamDecorator(
+  (data: string, ctx: ExecutionContext) => {
+    const request = ctx.switchToHttp().getRequest();
+    return request.user.userId;
+  },
+);
+
+/**
+ * @return all data from jwt payload
+ */
+export const UserInfo = createParamDecorator(
+  (data: string, ctx: ExecutionContext) => {
+    const request = ctx.switchToHttp().getRequest();
+    const userInfo: IJwtPayload = {
+      userId: request.user.userId,
+      phone: request.user.phone,
+      username: request.user.username,
+      logInTime: request.user.logInTime,
+    };
```

Comment:
simply return request.user so that if any parameter is added in token then we dont need to update that here